File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.go
/**
* Definition for singly-linked list.
* type ListNode struct {
* Val int
* Next *ListNode
* }
*/

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	root := &ListNode{-1, head}
	tail := root
	
	for (n > 1) {head = head.Next; n--}
	for (head.Next != nil) {head, tail = head.Next, tail.Next}
	
	tail.Next = tail.Next.Next
	
	return root.Next
}

File: LeetCode Solutions/19. Remove Nth Node From End of List/solution19.php
<?php 
  /**
  * Definition for a singly-linked list.
  * class ListNode {
  *   public $val = 0;
  *   public $next = null;
  *   function __construct($val = 0, $next = null) {
  *     $this->val = $val;
  *     $this->next = $next;
  *   }
  * }
  */
  class Solution {
  /**
  * @param ListNode $head
  * @param Integer $n
  * @return ListNode
  */
    function removeNthFromEnd($head, $n) {
      $root = new ListNode(0, $head); $tail = $root;
      
      while ($n > 1) {$head = $head -> next; $n--;}
      while ($head->next != NULL) {
        $head = $head -> next;
        $tail = $tail -> next;
      }
      
      $tail -> next = $tail -> next -> next;
      
      return $root -> next;
    }
  }
?>